### PR TITLE
Bootstrapping the work on issue #14, without serde_utils.

### PR DESCRIPTION
**Pull request by Yoric**
Variant on pull request #16, without the dependency on serde_utils.

### CODE REVIEW

File: src/values.rs [hunk 1]
```diff
@@ -109,9 +133,50 @@ impl Ord for OnOff {
...
+            OnOff::On => "On".serialize(serializer),
+            OnOff::Off => "Off".serialize(serializer)
+        }
+    }
+}
+impl Deserialize for OnOff {
+    fn deserialize<D>(deserializer: &mut D) -> Result<Self, D::Error> where D: Deserializer {
+        deserializer.deserialize_string(TrivialEnumVisitor::new(|source| {
+            match source {
+                "On" | "on" | "ON" => Ok(OnOff::On),
```
Comment (azasypkin) [inline src/values.rs]: **question:** I have a mixed feelings on this, from one side this approach looks a bit fragile (e.g. for multiple-words variants like `MotionDetected` it's not that obvious what formats we should support), from the other one good error message should help developer to figure out if format he's using is wrong. Do you know if something like `source.to_ascii_upper()` have too big memory impact for our use cases?

Comment (Yoric) [inline src/values.rs]: I don't think that there will be any meaningful impact, but I realize that we probably shouldn't do that just for some deserializers and not for others. So let's roll back to keeping only the documented format, for the time being, and then, if we find a way to make it work for all enumerations, consider generalizing to case variations?

Does this sound good to you?

Comment (azasypkin) [inline src/values.rs]: Yep, sounds good to me.

**Comment by azasypkin** [comment]
Personally I like this PR more than the one with `serde_utils`. Right now we use it only for the single case, so no need to introduce hard dependency to external crate, otherwise more and more places can start depend on it and eventually it will be too hard to get rid of it :)

So **r+**  (be aware of my still limited Rust knowledge though :))! Just a small trailing-comma nit from previous PR + link to serde enum serialization issue to get more background.

**Comment by Yoric** [comment]
Thanks for the review!

**Pull request merged**